crates/cxx-qt-gen/src/parser/mod.rs

```diff
@@ -25,9 +29,67 @@ use syn::{
     punctuated::Punctuated,
     spanned::Spanned,
     token::{Brace, Semi},
-    Error, Ident, Item, ItemMod, Meta, Result, Token,
+    Attribute, Error, ForeignItemFn, Ident, Item, ItemMod, Meta, Result, Token,
 };
 
+fn check_safety(method: &ForeignItemFn, safety: &Safety) -> Result<()> {
+    if safety == &Safety::Unsafe && method.sig.unsafety.is_none() {
+        Err(Error::new(
+            method.span(),
+            "Invokables must be marked as unsafe or wrapped in an `unsafe extern \"RustQt\"` block!",
+        ))
+    } else {
+        Ok(())
+    }
+}
+
+pub trait Invokable {
+    fn name(&self) -> &Name;
+}
+
+/// Struct with common fields between Invokable types.
+/// These types are ParsedSignal, ParsedMethod and ParsedInheritedMethod
+pub struct MethodFields {
+    qobject_ident: Ident,
+    mutable: bool,
+    parameters: Vec<ParsedFunctionParameter>,
+    safe: bool,
+    name: Name,
+    docs: Vec<Attribute>, // TODO: Remove this
+}
```

Comment:
Quite nice to see this being deduplicated.
I would prefer to move this type and associated functions into its own file though. You could probably also just put it into method.rs. But it doesn't really belong into the parser module.

Reply:
Moved into method.rs